### ISSUE TITLE
API search `Computer_Item` has empty result with `forcedisplay[0]=3`

### ISSUE THREAD

**Issue by aagz** [glpi-project/glpi#7278]
**Describe the bug**
API
Empty search result of `Computer_Item` if exists parameter `forcedisplay[0]=3`

**Page(s) URL**

URL: `/apirest.php/search/Computer_Item/?forcedisplay[0]=3`
Result: 
```JSON
{
  "totalcount": null,
  "count": 0,
  "sort": 1,
  "order": "ASC",
  "content-range": "0--1/"
}
```

**To reproduce**

1. Send query to API
2. See error

**Expected behavior**

if change `forcedisplay[0]=3` to `forcedisplay[0]=2` we get:
```JSON
{
  "totalcount": 3,
  "count": 3,
  "sort": 1,
  "order": "ASC",
  "data": [
    {
      "2": 2
    },
    {
      "2": 5
    },
    {
      "2": 6
    }
  ],
  "content-range": "0-2/3"
}
```

if change to any other number e.g. 324:
```JSON
{
  "totalcount": 3,
  "count": 3,
  "sort": 1,
  "order": "ASC",
  "data": [
    [],
    [],
    []
  ],
  "content-range": "0-2/3"
}
```

**Screenshots**

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
{
    "glpi": {
        "uuid": "9lFpTG7Dmw6BZk1Lfz1LenZ1Qml4mskVREKfccVG",
        "version": "9.4.4",
        "plugins": [
            {
                "key": "fusioninventory",
                "version": "9.4+2.4"
            },
            {
                "key": "myplu",
                "version": "0.0.1"
            },
            {
                "key": "example",
                "version": "7.1"
            },
            {
                "key": "solvecloseticketaction",
                "version": "9.4+1.0"
            }
        ],
        "default_language": "en_GB",
        "install_mode": "TARBALL",
        "usage": {
            "avg_entities": "0-500",
            "avg_computers": "0-500",
            "avg_networkequipments": "0-500",
            "avg_tickets": "0-500",
            "avg_problems": "0-500",
            "avg_changes": "0-500",
            "avg_projects": "0-500",
            "avg_users": "0-500",
            "avg_groups": "0-500",
            "ldap_enabled": true,
            "mailcollector_enabled": false,
            "notifications_modes": [],
            "notifications": [
                "mailing",
                "ajax"
            ]
        }
    },
    "system": {
        "db": {
            "engine": "(Ubuntu)",
            "version": "5.7.30-0ubuntu0.18.04.1",
            "size": "52.7",
            "log_size": "",
            "sql_mode": ""
        },
        "web_server": {
            "engine": "Apache",
            "version": "2.4.29"
        },
        "php": {
            "version": "7.2.24",
            "modules": [
                "Core",
                "date",
                "libxml",
                "openssl",
                "pcre",
                "zlib",
                "filter",
                "hash",
                "Reflection",
                "SPL",
                "sodium",
                "session",
                "standard",
                "apache2handler",
                "mysqlnd",
                "PDO",
                "xml",
                "apcu",
                "apc",
                "calendar",
                "ctype",
                "curl",
                "dom",
                "mbstring",
                "fileinfo",
                "ftp",
                "gd",
                "gettext",
                "iconv",
                "imap",
                "json",
                "ldap",
                "exif",
                "mysqli",
                "pdo_mysql",
                "Phar",
                "posix",
                "readline",
                "shmop",
                "SimpleXML",
                "sockets",
                "sysvmsg",
                "sysvsem",
                "sysvshm",
                "tokenizer",
                "wddx",
                "xmlreader",
                "xmlrpc",
                "xmlwriter",
                "xsl",
                "Zend OPcache",
                "xdebug"
            ],
            "setup": {
                "max_execution_time": "30",
                "memory_limit": "128M",
                "post_max_size": "8M",
                "safe_mode": false,
                "session": "files",
                "upload_max_filesize": "2M"
            }
        },
        "os": {
            "family": "Linux",
            "distribution": "",
            "version": "4.15.0-74-generic"
        }
    }
}
```

**Additional context**

`/listSearchOption/Computer_Item` has the parameter.
```JSON
{
  "common": {
    "name": "Characteristics"
  },
  "2": {
    "name": "ID",
    "table": "glpi_computers_items",
    "field": "id",
    "datatype": "number",
    "nosearch": false,
    "nodisplay": false,
    "available_searchtypes": [
      "contains",
      "notcontains"
    ],
    "uid": "Computer_Item.id"
  },
  "3": {
    "name": "Computers",
    "table": "glpi_computers",
    "field": "computers_id",
    "datatype": "text",
    "nosearch": false,
    "nodisplay": false,
    "available_searchtypes": [
      "contains",
      "notcontains"
    ],
    "uid": "Computer_Item.Computer.computers_id"
  }
}
```

**Comment by cedric-anne** [comment]
Hi,

Could you try with patch #7456 applied ?

Regards

**Comment by cedric-anne** [comment]
Should be fixed by #7456
